junit-jupiter-api/src/main/java/org/junit/jupiter/api/AssertDoesNotThrow.java

```diff
@@ -49,7 +50,7 @@ private static void assertDoesNotThrow(Executable executable, Object messageOrSu
 		}
 		catch (Throwable t) {
 			String message = buildPrefix(nullSafeGet(messageOrSupplier)) + "Unexpected exception thrown: "
```

Comment:
This line `String message = buildPrefix(nullSafeGet(messageOrSupplier)) + "Unexpected exception thrown: "` is duplicated below. Merge both into a new private `createMessage(messageOrSupplier, t)` or even a `createAssertionFailedError(messageOrSupplier, t)`?